src/welcome.rs

```diff
@@ -126,6 +128,90 @@ pub(crate) fn assign_talk_role(cx: &Context, reaction: &Reaction) -> Result<(),
     Ok(())
 }
 
+pub(crate) fn update_welcome_message(args: Args) -> Result<(), Error> {
+    use std::str::FromStr;
+
+    let new_message = &args
+        .params
+        .get("message")
+        .ok_or("unable to retrieve message param")?;
+
+    let welcome_message_is_not_cached = {
+        let data = args.cx.data.read();
+        !data.contains::<CachedWelcomeMessage>()
+    };
+
+    if welcome_message_is_not_cached {
+        info!("Welcome message not cached, caching");
+        let conn = DB.get()?;
+
+        let res = conn
+            .build_transaction()
+            .read_only()
+            .run::<_, Box<dyn std::error::Error>, _>(|| {
+                let res: Option<_> = messages::table
+                    .filter(messages::name.eq("welcome"))
+                    .first::<(i32, String, String, String)>(&conn)
+                    .optional()?;
+
+                Ok(res)
+            })?;
+
+        if let Some((_, _, message_id, channel_id)) = res {
+            let message = ChannelId::from(u64::from_str(&channel_id)?)
+                .message(args.cx, u64::from_str(&message_id)?)?;
+
+            cache_welcome_message(args.cx, message)?;
+        } else {
```

Comment:
Instead of failing, maybe we should somehow have it post the message? Might be tricky, ergonomically speaking, because of the lack of awareness of which channel to post in.